db/sequence_allocator.go

```diff
@@ -306,6 +326,20 @@ func (s *sequenceAllocator) _reserveSequenceBatch(ctx context.Context) error {
 		return err
 	}
 
+	// check for rollback of _sync:seq document
+	expectedValue := s.max + s.sequenceBatchSize
+	if max < expectedValue {
+		// rollback of _sync:seq detected
+		base.ErrorfCtx(ctx, "rollback of _sync:seq document detected. Allocated to %d but expected value of at least %d", max, expectedValue)
+		// find diff between current _sync:seq value and what we expected it to be + correction value
+		correctionIncrValue := (expectedValue - max) + syncSeqCorrectionValue
```

Comment:
It feels like these four lines (332-335) should be moved into fixSyncSeqRollback (and the same for other uses).